RocketChat/Rocket.Chat
# fix: Custom OAuth roles are not correctly synced on user creation

## Proposed changes (including videos or screenshots)

 - Add role sync step on OAuth user creation.

## Issue(s)


## Steps to test or reproduce

Configure a custom OAuth service with "Merge Roles from SSO" enabled and add some roles to sync. Use the custom OAuth service to login for the first time with a user that has any of the roles to sync assigned to them.
**_Current behavior:_** the roles are only assigned on their second login.
**_Expected behavior:_** the user should have their roles synced since the first login, on user creation.

## Further comments

[SUP-544](https://rocketchat.atlassian.net/browse/SUP-544)

[SUP-544]: https://rocketchat.atlassian.net/browse/SUP-544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
